iothub/device/samples/PnpDeviceSamples/TemperatureController/Program.cs

```diff
@@ -0,0 +1,327 @@
+    {
+        private const string ModelId = "dtmi:com:example:TemperatureController;1";
+        private const string Thermostat1 = "thermostat1";
+        private const string Thermostat2 = "thermostat2";
+        private const string SerialNumber = "SR-123456";
+
+        private static readonly string s_deviceConnectionString = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONNECTION_STRING");
+        private static readonly Random s_random = new Random();
+
+        private static DeviceClient s_deviceClient;
+        private static ILogger s_logger;
+
+        // A dictionary to hold all desired property change callbacks that this pnp device should be able to handle.
+        // The key for this dictionary is the componentName.
+        private static readonly IDictionary<string, DesiredPropertyUpdateCallback> s_desiredPropertyUpdateCallbacks = new Dictionary<string, DesiredPropertyUpdateCallback>();
+
+        // Dictionary to hold the temperature updates sent over each "Thermostat" component.
+        private static readonly Dictionary<string, Dictionary<DateTimeOffset, double>> s_temperatureReadings = new Dictionary<string, Dictionary<DateTimeOffset, double>>();
+
+        // Dictionary to hold the current temperature for each "Thermostat" component.
+        private static readonly Dictionary<string, double> s_temperature = new Dictionary<string, double>();
+
+        // Dictionary to hold the max temperature since last reboot, for each "Thermostat" component.
+        private static readonly Dictionary<string, double> s_maxTemp = new Dictionary<string, double>();
+
+
+        public static async Task Main(string[] _)
+        {
+            ILoggerFactory loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder
+                .AddFilter(level => level >= LogLevel.Debug)
+                .AddConsole(options =>
+                {
+                    options.TimestampFormat = "[MM/dd/yyyy HH:mm:ss]";
+                });
+            });
+            s_logger = loggerFactory.CreateLogger<Program>();
+
+            await RunSampleAsync();
+        }
+
+        private static async Task RunSampleAsync()
+        {
+            // This sample follows the following workflow:
+            // -> Initialize device client instance.
+            // -> Set handler to receive "reboot" command - root interface.
+            // -> Set handler to receive "getMaxMinReport" command - on "Thermostat" components.
+            // -> Set handler to receive "targetTemperature" property updates from service - on "Thermostat" components.
+            // -> Send initial device info - "workingSet" over telemetry, "serialNumber" over reported property update - root interface.
+            // -> Periodically send "temperature" over telemetry - on "Thermostat" components.
+            // -> Send "maxTempSinceLastReboot" over property update, when a new max temperature is set - on "Thermostat" components.
```

Comment:
on line 75; why not immediately?

Reply:
It is sent immediately after device starts, and then every 5secs.